File: Server/unblockBuddy.php
<?php
  require "globalFunctions.php";
  $pdo = new PDO("pgsql:host=".HOST.";port=".PORT.";dbname=".DATABASE.";user=".USERNAME.";password=".PASSWORD);

  $myId = $_POST["myId"];
  $token = $_POST["token"];
  $buddyId = $_POST["buddyId"];

	$isValid =
		isAuthenticated($myId, $token) &&
		isExtantUserId($buddyId);
	if (!$isValid) {
  	$pdo = null;
		die("Invalid");
	}

  $query = "delete from block where user_id = ? and buddy_id = ?;";
  $stmt = $pdo->prepare($query);
  $stmt->execute(array($myId, $buddyId));

	$pdo = null;
?>

File: Server/unblockBuddy.go
package main

import (
	"context"
	"fmt"
	"net/http"
)

func unblockBuddy(w http.ResponseWriter, r *http.Request) {
	r.ParseForm()
	token := r.Form.Get("token")
	buddyId := r.Form.Get("buddyId")
	myId := getMyId(token)

	isValid :=
		isAuthenticated(myId) &&
			isExtantUserId(buddyId)
	if !isValid {
		fmt.Fprintln(w, "Invalid")
		return
	}

	_, err := pool.Exec(context.Background(), "delete from block where user_id = $1 and buddy_id = $2;", myId, buddyId)
	handleError(err)
}
